src/components/markdown/code-block.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { CopyIcon } from '../icons/copy';
+import copy from 'copy-to-clipboard';
+import styled from 'styled-components';
+
+interface CodeBlockProps {
+  children: React.ReactNode;
+}
+
+const StyledCodeBlock = styled.pre`
+  padding: 1em;
+  margin: 0.5em 0;
+  overflow: auto;
+  position: relative;
+`;
+
+const CodeBlockWrapper = styled.div`
+  background: #f5f2f0;
+  position: relative;
+  cursor: default;
+
+  &:hover {
+    background: #e4e8f3;
+  }
+
+  > svg {
+    position: absolute;
+    top: 16px;
+    right: 16px;
+    cursor: pointer;
+    z-index: 2;
+    &:active {
+      border: 2px solid #f5f2f0;
+    }
+    &:hover {
+      > circle {
+        fill: #668cff;
+      }
+    }
+  }
+`;
+
+export const CodeBlock = (props: CodeBlockProps) => {
+  const copyCodeBlock = (event) => {
+    const codeNode = event.currentTarget.nextSibling;
+    const { innerText } = codeNode;
+    copy(innerText);
```

Comment:
that's pretty indirect, can't we explicitly query for the  `pre` tag ? This currently relies on a fixed DOM order of the icon followed by the pre tag.

![Screen Shot 2021-08-31 at 12 01 17](https://user-images.githubusercontent.com/1701755/131483344-763e946b-383c-437b-9662-82f286f05d3c.png)